src/System.Runtime.CompilerServices.Unsafe/Directory.Build.props

```diff
@@ -1,10 +1,11 @@
 <Project>
 
-  <Import Project="..\Directory.Build.props" />
+  <Import Project="$([MSBuild]::GetPathOfFileAbove(Directory.Build.props, $(MSBuildThisFileDirectory)..))" />
 
   <PropertyGroup>

```

Comment:
@ericstj / @carlossanlop I wonder if it would be better to keep these version properties in the src project (for consistency with every other lib in this repo) and provide them to the P2P (Versioning.csproj) via global properties. That would also allow reusing the Versioning.csproj if we ever hit another case like this one.

Btw, Versioning.csproj should probably be renamed to Versioning.proj and not use The Microsoft.NET.Sdk but the Microsoft.Build.NoTargets as the project shouldn't invoke the compiler.

Reply:
It needs to invoke the compiler to produce an assembly.  The assembly is then disassembled for its native version resources.

It is specific to System.Runtime.CompilerServices.Unsafe.  It cannot be shared.

Reply:
> It needs to invoke the compiler to produce an assembly. The assembly is then disassembled for its native version resources.

Interesting. I didn't know that roslyn by default allows to compile an assembly without any compile files included.

> It is specific to System.Runtime.CompilerServices.Unsafe. It cannot be shared.

Given that it doesn't have any properties nor items, it could be shared. The needed state could just be passed in via global properties.

Reply:
> it could be shared. The needed state could just be passed in via global properties.

I'm not sure what you meant by this. Do you mean moving the `<Import>` line to its own file in the parent directory?

Reply:
@ericstj explained it. I opened an issue to track this: https://github.com/dotnet/maintenance-packages/issues/108

Reply:
Honestly, I don't think it's worth it. I just wanted to mention it.